// restaurantorders/index.js
var orders = [{
  code: 1,
  client: 'Guido La Vespa',
  description: '2 carne e 1 carbonara',
  bill: 51.00,
  status: 'new'
},
{
  code: 2,
  client: 'Claudia belli',
  description: '1 braciola e 2 stigghiola',
  bill: 30.00,
  status: 'ready'
}, {
  code: 3,
  client: 'Mario biondi',
  description: '1 pesce spada e 1 amatriciana',
  bill: 35.00,
  status: 'closed'
}]

var newOrders = [];


exports.getOrders = function() {
  return orders;
}

exports.getOrdersbyStatus = function(status) {
  for (var index in orders){
    var Orders = orders[index]
    if (Orders.status === status) {
     return newOrders.push(Orders)
    }
  }
  return null;

}

exports.getProfit = function(){
  var profit = 0;
  for (var index in orders) {
     profit += orders[index].bill  ;
  }
  return profit;
}

exports.addOrder = function(order){
      orders.push({
  code: order.code,
  client : order.client,
  description: order.description,
  bill: order.bill,
  status: order.status
});
return null;
}

exports.setOrder = function(code, newStatus) {
  for (var index in orders) {
    var order = orders[index];
    if(code === order.code){
   order.status = newStatus;
 }
  }
  return null;
}
exports.getOrdersN = function (status) {
  for(var index in orders) {
    var order = orders[index];
      if(code === order.code){
   return newOrders.push({
    code: order.code,
    client: order.client,
    description:  order.description,
    bill: order.bill,
    status: order.status
    })
  }
}

exports.getOrdersR = function (status) {
  for(var index in orders) {
    var readyOrders = orders[index];
   return readyOrders.push({
    code: order.code,
    client: order.client,
    description:  order.description,
    bill: order.bill,
    status: order.status
    })
  }
}
exports.deleteOrder = function(code) {
    for(var i=0; i<orders.length; i++){
      if (orders[i].code === code) {
          return orders.splice(i, 1);
      }
    }
}
