// ecom-products/index.js
var products = [{
  name: 'astuccio',
  price: 5,
  quantity: 7,
  id:1,
  sold: 0
},
{
  name: 'zaino',
  price: 9,
  quantity: 12,
  id:2,
  sold: 0
}, {
  name: 'KimJon',
  price: 4,
  quantity: 7,
  id:3,
  sold: 0
},
{
  name: 'pennapilot',
  price: 4,
  quantity: 7,
  id:4,
  sold: 0
},
{
  name: 'matita',
  price: 1,
  quantity: 8,
  id:5,
  sold: 0
}]

var soldProducts = [];
var soldProductsbyToken = [];
var backup = JSON.parse(JSON.stringify(products));

exports.all = function(){
  return products;
}

exports.getById = function(id){
  for (var index in products) {
    var product = products[index];
    if (product.id === id) {
      return product;
    }
  }
  return null;
}

exports.addProduct = function(newProduct){
  var lastProductId = products[products.length-1].id;
  products.push({name: newProduct.name,
                 price: newProduct.price,
                 quantity: newProduct.quantity,
                 id: lastProductId+1
               })
}

exports.deleteProduct = function(id){
  for (var index in products) {
    var product = products[index];
    if (product.id == id) {
      return products.splice(index, 1);
    }
  }
  return null;
}

exports.updateProduct = function(id, newProduct){
  for (var index in products) {
    var product = products[index];
    if (products[index].id == id) {
      products[index].name = newProduct.name;
      products[index].price = newProduct.price;
      products[index].quantity = newProduct.quantity;
      return products[index];
    }
  }
  return null;
}

exports.reset = function() {
    products = JSON.parse(JSON.stringify(backup));
}

exports.buy = function(id, newBuyer){

    for (var index in products) {
      var product = products[index];
        if (product.quantity < 1 ){
          return "Out of stock, you can still choose another product to buy";
        }
       if (product.id == id) {
        product.quantity = product.quantity - 1;
        product.sold = product.sold + 1;
        var buyerName = newBuyer;
        soldProducts.push({
          buyerName: buyerName,
          productID: id,
          dateofPurchase: new Date().toString()
        })
        return product;
      }
    }
}

exports.purchases = function(){
    return soldProducts;
}

exports.purchasesByToken = function(buyerName){
  for (var index in soldProducts) {
    var sales = soldProducts[index];
    if(soldProducts.buyerName == buyerName) {
      soldProductsbyToken.push(sales)

    }
}
  return soldProductsbyToken;
}
